src/components/charts/SensorDetailsBarChart.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable react/require-default-props */
+import { ChartData, ChartOptions } from "chart.js";
+import CountSensorSchema from "../models/readings/CountSensorSchema";
+import SensorReadingSchema from "../models/readings/SensorSchema";
+import { getFormattedCountData } from "../presentation/uiHelpers";
+import { CHART_DATE_FORMAT } from "./chartHelper";
+import BarChart from "./BarChart";
+
+type SensorDetailsBarChartProps = {
+  label: string;
+  chartColor: string;
+  data: {
+    when: string;
+    value: number;
+  }[];
+  schema: SensorReadingSchema<number>;
+};
+
+export function getTooltipDisplayText(
```

Comment:
Similar comment here: you could probably move this function to `chartHelper` and have it handle all schema types to cut down on some duplicate code.

Reply:
Done.